File: back/src/models/Visitation.js
import mongoose from 'mongoose';

const visitationSchema = new mongoose.Schema(
  {
    servantUsername: { type: String, required: true, index: true },
    makhdoomId: { type: String, required: true, index: true },
    weekDate: { type: String, required: true, index: true },
    result: { type: String, default: 'answered' },
    notes: { type: String, default: '' },
    scannedAt: { type: String, default: () => new Date().toISOString() },
    serviceYear: { type: String, default: '2026', index: true },
  },
  {
    timestamps: true,
    strict: false,
    collection: 'servantVisitations',
  }
);

visitationSchema.index({ servantUsername: 1, makhdoomId: 1, weekDate: 1 });

const Visitation = mongoose.models.Visitation || mongoose.model('Visitation', visitationSchema);
export default Visitation;

File: back/Models/Visitation.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Eftekad.Backend.Models
{
    [Table("servantVisitations")]
    public class Visitation
    {
        [Key]
        [Column("id")]
        public string Id { get; set; } = string.Empty;

        [Column("servantUsername")]
        public string ServantUsername { get; set; } = string.Empty;

        [Column("makhdoomId")]
        public string MakhdoomId { get; set; } = string.Empty;

        [Column("weekDate")]
        public string WeekDate { get; set; } = string.Empty;

        [Column("result")]
        public string Result { get; set; } = "answered";

        [Column("notes")]
        public string Notes { get; set; } = string.Empty;

        [Column("scannedAt")]
        public string ScannedAt { get; set; } = DateTime.UtcNow.ToString("o");

        [Column("serviceYear")]
        public string ServiceYear { get; set; } = "2026";

        [Column("createdAt")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [Column("updatedAt")]
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
}
